checkov/terraform/checks_infra/solvers/complex_solvers/base_complex_solver.py

```diff
@@ -1,28 +1,31 @@
+from typing import List, Any, Tuple, Dict
+
+from networkx import DiGraph
+
 from checkov.common.graph.checks_infra.enums import SolverType
 from checkov.common.graph.checks_infra.solvers.base_solver import BaseSolver
 
 
 class BaseComplexSolver(BaseSolver):
-    operator = ''
+    operator = ""
 
-    def __init__(self, solvers, resource_types):
+    def __init__(self, solvers: List[BaseSolver], resource_types: List[str]) -> None:
         if solvers is None:
             solvers = []
         self.solvers = solvers
         self.resource_types = resource_types
         super().__init__(SolverType.COMPLEX)
 
-    def _get_operation(self, *args, **kwargs):
+    def _get_operation(self, *args: Any, **kwargs: Any) -> Any:
         raise NotImplementedError()
 
-    def _get_negative_op(self, *args):
+    def _get_negative_op(self, *args: Any) -> bool:
```

Comment:
@gruebel Pretty sure this doesn't return true / false

Reply:
yeah, you are right 😄